### PR TITLE
Altered attribute type checking for different Integer and float types

### PR DESCRIPTION
**Pull request by rohan-panchal**
Changed the flow so that the different integer types Integer16,
Integer32, Integer64 would be handled in a way that non 64-bit
architectures could handle them correctly.

(First open source contribution don't kill me)

### CODE REVIEW

File: Classes/NSManagedObject+ActiveRecord.m [hunk 1]
```diff
@@ -366,17 +366,23 @@ - (void)setSafeValue:(id)value forKey:(NSString *)key {
 
     else if ([value isKindOfClass:[NSString class]]) {
 
-        if ([self isIntegerAttributeType:attributeType])
```
Comment (stephencelis) [inline Classes/NSManagedObject+ActiveRecord.m]: What if we keep this, but change `numberWithInteger`, below to be `numberWithLongLong`. That should satisfy 32-bit systems :+1:

File: Classes/NSManagedObject+ActiveRecord.m [hunk 2]
```diff
@@ -366,17 +366,23 @@ - (void)setSafeValue:(id)value forKey:(NSString *)key {
...
-
-        else if (attributeType == NSBooleanAttributeType)
-            value = [NSNumber numberWithBool:[value boolValue]];
-
-        else if (attributeType == NSFloatAttributeType)
-            value = [NSNumber numberWithDouble:[value doubleValue]];
-
-        else if (attributeType == NSDateAttributeType)
-            value = [self.defaultFormatter dateFromString:value];
+      if (attributeType == NSInteger16AttributeType)
```
Comment (stephencelis) [inline Classes/NSManagedObject+ActiveRecord.m]: Can you please maintain the 4-space indentation of the file?

**Comment by stephencelis** [comment]
Thanks for the PR! And congrats on your first open source contribution :smile:

Left a few comments. Could you please push some fixes and we should be good to merge!

File: Classes/NSManagedObject+ActiveRecord.m (hunk 1 above)
Comment (rohan-panchal) [inline Classes/NSManagedObject+ActiveRecord.m]: numberWithLongLong should work. i'll change/test that.

File: Classes/NSManagedObject+ActiveRecord.m (hunk 2 above)
Comment (rohan-panchal) [inline Classes/NSManagedObject+ActiveRecord.m]: Yup. i'll change that.

File: Classes/NSManagedObject+ActiveRecord.m [hunk 3]
```diff
@@ -366,17 +366,23 @@ - (void)setSafeValue:(id)value forKey:(NSString *)key {
 
     else if ([value isKindOfClass:[NSString class]]) {
 
-        if ([self isIntegerAttributeType:attributeType])
-            value = [NSNumber numberWithInteger:[value integerValue]];
-
-        else if (attributeType == NSBooleanAttributeType)
-            value = [NSNumber numberWithBool:[value boolValue]];
-
-        else if (attributeType == NSFloatAttributeType)
```
Comment (stephencelis) [inline Classes/NSManagedObject+ActiveRecord.m]: In fact, looks like we're missing `NSDoubleAttributeType`, too. Let me know if you want to add a check for that!

Comment (rohan-panchal) [inline Classes/NSManagedObject+ActiveRecord.m]: It seems that the 64 bit changes don't yield any specific caviats. I can add that check as well.

**Comment by stephencelis** [comment]
Thanks! :boom:

**Pull request merged**